python/hsfs/engine/spark.py

```diff
@@ -275,3 +275,34 @@ def _setup_s3(self, storage_connector, path):
 
 class SchemaError(Exception):
     """Thrown when schemas don't match"""
+
+
+# Replace Pydoop split method to be able to support hopsfs:// schemes
+class _HopsFSPathSplitter(hdfs.path._HdfsPathSplitter):
+    import os
+    import re
+    import time
+
+    @classmethod
+    def split(cls, hdfs_path, user):
+        if not hdfs_path:
+            cls.raise_bad_path(hdfs_path, "empty")
+        scheme, netloc, path = cls.parse(hdfs_path)
+        if not scheme:
+            scheme = "file" if hdfs_fs.default_is_local() else "hdfs"
+        if scheme == "hdfs" or scheme == "hopsfs":
+            if not path:
+                cls.raise_bad_path(hdfs_path, "path part is empty")
+            if ":" in path:
+                cls.raise_bad_path(hdfs_path, "':' not allowed outside netloc part")
+            hostname, port = cls.split_netloc(netloc)
+            if not path.startswith("/"):
+                path = "/user/%s/%s" % (user, path)
+        elif scheme == "file":
+            hostname, port, path = "", 0, netloc + path
+        else:
+            cls.raise_bad_path(hdfs_path, "unsupported scheme %r" % scheme)
+        return hostname, port, path
+
+
+hdfs.path._HdfsPathSplitter = _HopsFSPathSplitter
```

Comment:
Did you test this? We are always importing this module in the __init__.py of the engine package, so I would think it will fail on SageMaker. I think the class is fine to have in this module, but we should move this line to be executed only when we have pydoop available. Or we make pydoop a dependency so it's also available on other environments.

Reply:
I see your point. No, we can't make Pydoop available everywhere, as installing Pydoop requires having an Hadoop distribution available. 

Maybe I can move the initialization in the constructor so it runs only if the Spark engine is actually created.

Reply:
yes either that, or moving it to this if-clause here, https://github.com/logicalclocks/feature-store-api/blob/eb1312283d12ea9e0df1a864187df2c3678a168c/python/hsfs/engine/__init__.py#L26